photon-client/src/main/scala/com/linkedin/photon/ml/data/avro/NameAndTermFeatureMapUtils.scala

```diff
@@ -0,0 +1,153 @@
+/*
+ * Copyright 2017 LinkedIn Corp. All rights reserved.
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may
+ * not use this file except in compliance with the License. You may obtain a
+ * copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations
+ * under the License.
+ */
+package com.linkedin.photon.ml.data.avro
+
+import org.apache.hadoop.fs.{FileSystem, FileUtil, Path}
+import org.apache.spark.SparkContext
+import org.apache.spark.rdd.RDD
+
+/**
+ * Some basic functions to deal with NameAndTermFeatureMap: a [[Map]] of feature section key to [[NameAndTerm]] feature
+ * [[RDD]].
+ *
+ */
+object NameAndTermFeatureMapUtils {
+
+  /**
+   * Get the map from feature name of type [[NameAndTerm]] to feature index of type [[Int]] based on the specified
+   * feature section keys from the input.
+   *
+   * @param nameAndTermFeatureMap The [[Map]] of feature section key to [[NameAndTerm]] feature [[RDD]]
+   * @param featureSectionKeys The specified feature section keys to generate the name to index map explained above
+   * @param isAddingIntercept Whether to add a dummy variable to the generated feature map to learn an intercept term
+   * @param sc The Spark context
+   * @return The generated map from feature name of type [[NameAndTerm]] to feature index of type [[Int]]
+   */
+  def getFeatureNameAndTermToIndexMap(
+      nameAndTermFeatureMap: Map[String, RDD[NameAndTerm]],
+      featureSectionKeys: Set[String],
+      isAddingIntercept: Boolean,
+      sc: SparkContext): Map[NameAndTerm, Int] = {
+
+    val featureNameAndTermToIndexMap = nameAndTermFeatureMap
+      .filterKeys(featureSectionKeys.contains)
+      .values
+      .fold(sc.emptyRDD[NameAndTerm])(_ ++ _)
```

Comment:
If replaced by reduce(_ ++ _), the GameTrainingDriverIntegTest. testFixedEffectInterceptOnlyFeatureBagsDir would fail with java.lang.UnsupportedOperationException: empty.reduceLeft, probably due to an empty nameAndTermFeatureMap after filterKeys.

Reply:
Ah, I see - this makes sense. If there's only one feature bag, then the second argument will be empty. Okay - understood. Please ignore this comment.